### PR TITLE
TASK-5979 - Automatize validation of SSO with CAS for OpenCGA Enterprise 2.x.x

### PR DESCRIPTION
**Pull request by pfurio**

### CODE REVIEW

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/AzureADAuthenticationManager.java [hunk 1]
```diff
@@ -420,4 +429,8 @@ public String createNonExpiringToken(String organizationId, String userId, Map<S
         throw new UnsupportedOperationException("Tokens are generated by Azure via authorization code or user-password");
     }
 
+    @Override
+    public void close() {
+        THREAD_POOL.shutdown();
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/AzureADAuthenticationManager.java:434]: :heart:

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java [hunk 2]
```diff
@@ -84,13 +85,18 @@ public JwtManager setPublicKey(Key publicKey) {
...
+    public String createJWTToken(String organizationId, AuthenticationOrigin.AuthenticationType type, String userId,
+                                 Map<String, Object> claims, long expiration) {
         long currentTime = System.currentTimeMillis();
 
         JwtBuilder jwtBuilder = Jwts.builder();
         if (claims != null && !claims.isEmpty()) {
             jwtBuilder.setClaims(claims);
         }
+        if (type != null) {
+            jwtBuilder.addClaims(Collections.singletonMap("authOrigin", type));
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java:97]: Not a constant here?

File: opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java [hunk 3]
```diff
@@ -216,6 +218,47 @@ public class UpdateNotesCommandOptions {
...
+        @Parameter(names = {"--include-result"}, description = "Flag indicating to include the created or updated document result in the response", required = false, help = true, arity = 0)
+        public boolean includeResult = false; 
+    
+        @Parameter(names = {"--authentication-origins"}, description = "Action to be performed if the array of authenticationOrigins is being updated.", required = false, arity = 1)
+        public String authenticationOrigins = "ADD"; 
+    
+        @Parameter(names = {"--optimizations-simplify-permissions"}, description = "The body web service simplifyPermissions parameter", required = false, help = true, arity = 0)
+        public boolean optimizationsSimplifyPermissions = false;
+    
+        @Parameter(names = {"--token-algorithm"}, description = "The body web service algorithm parameter", required = false, arity = 1)
```
Comment (j-coll) [inline opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java:251]: Missing param descriptions

**Review by j-coll — APPROVED** [review]

File: opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java (hunk 3 above)
Comment (pfurio) [reply opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java:251]: That's automatically generated. I know we need to change those description values :S

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java (hunk 2 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java:97]: Just created a constant :)

**Review by j-coll — DISMISSED** [review]

File: opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java (hunk 3 above)
Comment (j-coll) [reply opencga-app/src/main/java/org/opencb/opencga/app/cli/main/options/OrganizationsCommandOptions.java:251]: What I meant was that you should add these descriptions with "@DataField"  in the "Organization configuration"

**Review by j-coll — APPROVED** [review]

**Pull request merged**